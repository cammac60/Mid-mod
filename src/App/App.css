@import url('https://fonts.googleapis.com/css?family=Pacifico');

.App {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-image: url('../assets/cafe-img.jpeg');
  width: 100%;
  height: 100vh;
  background-repeat: no-repeat;
  background-size: cover;
  background-attachment: fixed;
  overflow-x: scroll;
  text-align: center;
}

.app-title {
  background-color: rgba(0, 5, 0, 0.8);
  color: teal;
  font-family: 'Pacifico', cursive;
  font-size: 50px;
}

.resy-container {
  display: grid;
  grid-gap: 30px;
  grid-template-columns: 1fr 1fr 1fr;
  margin-left: auto;
  margin-right: auto;
  width: 90%;
}

.resy-form {
  width: 100%;
}
