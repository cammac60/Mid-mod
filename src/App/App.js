import React, { Component } from 'react';
import './App.css';
import Card from '../Card/Card.js';

class App extends Component {
  constructor() {
    super();
    this.state = {
      reservations: []
    }
  }

  componentDidMount() {
    fetch('http://localhost:8000/api/v1/reservations')
    .then(response => response.json())
    .then(data => this.setState({reservations: [...this.state.reservations, data]}))
    .catch(err => console.log(err))
  }

  render() {
    let cards;
    if (this.state.reservations.length) {
      cards = this.state.reservations[0].map(res => {
        return ( <Card {...res} key={res.id}/> )
      });
    }
    return (
      <div className="App">
        <h1 className='app-title'>Turing Cafe Reservations</h1>
        <div className='resy-form'>

        </div>
        <div className='resy-container'>
          {cards}
        </div>
      </div>
    )
  }
}

export default App;
