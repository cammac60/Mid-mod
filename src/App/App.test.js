import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper, defaultState;

  beforeEach(() => {
    wrapper = shallow(<App />);
    defaultState = {
      reservations: [],
      name: '',
      date: '',
      time: '',
      number: ''
    }
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state upon mounting', () => {
    expect(wrapper.state()).toEqual(defaultState);
  });

  
});
