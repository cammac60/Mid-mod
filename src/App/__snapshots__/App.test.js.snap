// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="app-title"
  >
    Turing Cafe Reservations
  </h1>
  <div
    className="resy-form"
  >
    <Nav
      change={[Function]}
      click={[Function]}
      date=""
      name=""
      number=""
      reservations={Array []}
      time=""
    />
  </div>
  <div
    className="resy-container"
  />
</div>
`;
